frontend/packages/client/src/api/notificationService.js

```diff
@@ -0,0 +1,129 @@
+import {
+  LEANPLUM_APP_ID,
+  LEANPLUM_CRO_KEY,
+  LEANPLUM_EXPORT_KEY,
+} from 'api/constants';
+import Leanplum from 'leanplum-sdk';
+
+const COMMUNITY_UPDATES_CATEGORY_ID = 1;
+const options = {
+  method: 'GET',
+  headers: { accept: 'application/json' },
+};
+
+export const setUserId = async (walletId) => {
+  try {
+    Leanplum.setUserId(walletId);
+  } catch (e) {
+    throw new Error(e);
+  }
+};
+
+export const getUserSettings = async (walletId) => {
+  try {
+    let response = await fetch(
+      `https://api.leanplum.com/api?appId=${LEANPLUM_APP_ID}&clientKey=${LEANPLUM_EXPORT_KEY}&apiVersion=1.0.6&userId=${walletId}&action=exportUser`,
+      options
+    )
+    response = await response.json()
+    const data = response.response[0]
+    if (!data.success) {
+      throw new Error(data.error.message);
+    }
+    if (!data.userAttributes) {
+      throw new Error('User Not Found');
+    }
+
+    const communitySubscriptions = [];
+    const res = {
+      email: data.userAttributes.email,
+    };
+    let isSubscribedToCommunityUpdates = true;
+
+    for (const property in data.userAttributes) {
+      if (property.includes('community')) {
+        const communityId = property.split('community')[1];
+        communitySubscriptions.push({
+          communityId,
+          subscribed: data.userAttributes[property] === 'True',
+        });
+      }
+    }
+
+    res.communitySubscription = communitySubscriptions;
+
+    if (data.unsubscribeCategories) {
+      data.unsubscribeCategories.forEach((category) => {
+        if (parseInt(category.id) === COMMUNITY_UPDATES_CATEGORY_ID) {
+          isSubscribedToCommunityUpdates = false;
+        }
+      });
+    }
+
+    res.isSubscribedToCommunityUpdates = isSubscribedToCommunityUpdates;
+    return res;
+  } catch (e) {
+    throw new Error(e)
+  }
+};
+
+export const setUserEmail = async (email) => {
+  try {
+    await Leanplum.setUserAttributes({ email });
+    return true
+  } catch (e) {
+    throw new Error(e);
+  }
+};
+
+export const unsubscribeCommunity = async (communityId) => {
+  try {
+    Leanplum.setUserAttributes({ [`community${communityId}`]: false });
+    return true
+  } catch (e) {
+    throw new Error(e);
+  }
+};
+
+export const subscribeCommunity = async (communityId) => {
+  try {
+    Leanplum.setUserAttributes({ [`community${communityId}`]: true });
```

Comment:
as per the [doc](https://docs.leanplum.com/reference/user-attributes) only `string` and `number` type is supported, it looks like when fetch the data from Leanplum it will return as "True" ? https://github.com/DapperCollectives/CAST/pull/652/files#diff-57d84cd560a449e825c5114c1a3bf8924c84d9ea141de174a3b3555be6b020a3R48

Maybe we can explicitly store it as "true" to avoid implicit data conversion.

Reply:
it accepts this value but will return as a string, I have tested both cases and they both work

Reply:
i see, I would suggest setting the value explicitly to be a string as per their doc, that way we can expect the same value in get with no questions. (today Leanplum implicitly convert boolean to "True", we won't know how they will treat unsupported type in future).